### PR TITLE
OMotM: Fix crashes and oversights

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

File: config/formats.js [hunk 1]
```diff
@@ -689,7 +689,8 @@ let Formats = [
 			set.species = template.species;
 			return problems;
 		},
-		onModifyTemplate(template, target, effect) {
+		onModifyTemplate(template, target, source, effect) {
+			if (!effect || !target) return;
```
Comment (Zarel) [inline config/formats.js]: I'm pretty sure `!effect` is unnecessary.

Also how does Cross Evolution work? Should `/dt` just ignore it entirely?

Comment (KrisXV) [reply config/formats.js]: i dont really know how it works, its one of the few OMs i dont care enough to look into.

and yeah `!effect` is unnecessary, ill remove it

Comment (Zarel) [reply config/formats.js]: Aha, it works on nicknames which are unavailable in `/dt`. That makes sense.

Comment (KrisXV) [reply config/formats.js]: removed `!effect` @Zarel

File: config/formats.js [hunk 2]
```diff
@@ -722,8 +723,8 @@ let Formats = [
 		},
 		onBegin() {
 			let allPokemon = this.p1.pokemon.concat(this.p2.pokemon);
-			for (let i = 0, len = allPokemon.length; i < len; i++) {
-				allPokemon[i].baseTemplate = allPokemon[i].template;
+			for (const pokemon of allPokemon) {
```
Comment (Slayer95) [inline config/formats.js]:
```js
for (const pokemon of this.getAllPokemon()) {
// stuff
}
```

**Pull request merged**